### DISCUSSION TITLE
Importing users from Firebase Auth

### DISCUSSION THREAD

**Discussion by mukhtharcm** [pocketbase/pocketbase#5248] [Q&A]
Is there any way to import users from firebase auth into pocketbase?

I'm able to get these data out of every user from firebase auth

"localId
"email
"emailVerified
"passwordHash
"salt
"lastSignedInAt
"createdAt

**Comment by ganigeorgiev** [comment]
It would depend what algorithm is used for the password hashing (_you can find more information about your password settings in your Firebase console > Auth > Users > 3dots > "Password hash parameters"_).

PocketBase uses bcrypt. If your Firebase users are using `scrypt` then there is no direct way to import it in a PocketBase auth collection. One possible migration path could be to store the Firebase  password related data in a separate helper collection and use that that the first time the user tries to authenticate (eg. with the [`OnRecordBeforeAuthWithPasswordRequest`](https://pocketbase.io/docs/go-event-hooks/#onrecordbeforeauthwithpasswordrequest)). Once you verify that the user entered plain password matches the one from Firebase you can set it safely in the related user record with `user.SetPassword(pass)` and delete the fallback entry.

**Reply by mukhtharcm** [reply]
How to initially create users?
so that onRecordbeforeAuthWithPassworRequest will be ran?

I'm currently reading this while thinking I can do something about it

https://github.com/burggraf/supabase-user-password-migration

**Reply by mukhtharcm** [reply]
I think you deleted one of my message.
Sorry if it was a terrible solution.
And just FYI, I was not just copy pasting from chat gpt, but I did understand it for the most part.

Thank you

Sorry again for that :)

**Comment by mukhtharcm** [comment]
I think I'll just drop my plans for migration then.

The app was usable without an account. so it has almost 25k+ user generated content, but only have **38** registered users.
They have collectively created under 1.5k content.

I'll put a message on my website saying, everyone gotta create a new account, and if you need you content back, contact me and I'll do it manually.

wdyt?
@ganigeorgiev
